src/src/DataTypes/ESPEasyFileType.cpp

```diff
@@ -14,10 +18,27 @@ bool matchFileType(const String& filename, FileType::Enum filetype)
 
 bool isProtectedFileType(const String& filename)
 {
-  return matchFileType(filename, FileType::CONFIG_DAT) ||
-         matchFileType(filename, FileType::SECURITY_DAT) ||
-         matchFileType(filename, FileType::NOTIFICATION_DAT) ||
-         matchFileType(filename, FileType::PROVISIONING_DAT);
+  #if FEATURE_EXTENDED_CUSTOM_SETTINGS
+  bool isTaskSpecificConfig = false;
+  const String  fname       = filename.substring(filename.startsWith(F("/")) ? 1 : 0);
+  const String  mask        = F(DAT_TASKS_CUSTOM_EXTENSION_FILEMASK);
+  const uint8_t mPerc       = mask.indexOf('%');
```

Comment:
indexOf can return a -1 if not found. So you should not store the position in an unsigned type.